### ISSUE TITLE
Hyperparameter Optimization

### ISSUE THREAD

**Issue by Enzogall** [MathEpitech2025/my_torch#13]
Implement hyperparameter tuning mechanisms:

* Grid search for hyperparameters
* Learning rate scheduling
* Momentum and adaptive learning rates
* Document optimization results
